internal/QueryProcessor.py

```diff
@@ -55,79 +21,49 @@ class QueryProcessor:
     def query_google_vision(self, image):
-        # https://cloud.google.com/vision/docs/labels?hl=zh-cn#vision_label_detection-python
-        # Link for label detection (for example, identifying 'Coca-Cola')
-        client = vision.ImageAnnotatorClient()
         content = image.read()
         image = vision.Image(content=content)
-        response = client.label_detection(image=image)
+        response = self.vision_client.label_detection(image=image)
         labels = response.label_annotations
         vision_response = ', '.join([label.description for label in labels])
         return vision_response
 
     # TODO  System prompt will come from cognition if necessary
     def query_google_language(self, query: str):
-        # https://cloud.google.com/natural-language/docs/samples/language-entities-text?hl=zh_cn#language_entities_text-python
-        # Link for analyzing entities in a string (for example, returning information about 'Coca-Cola'). Not sure which Google Language API to use.
-        client = language_v2.LanguageServiceClient()
-        document_type_in_plain_text = language_v2.Document.Type.PLAIN_TEXT
-        language_code = "en"
         document = {
             "content": query,
-            "type_": document_type_in_plain_text,
-            "language_code": language_code,
+            "type_": language_v2.Document.Type.PLAIN_TEXT,
+            "language_code": "en",
         }
-        encoding_type = language_v2.EncodingType.UTF8
-        response = client.analyze_entities(
-            request={"document": document, "encoding_type": encoding_type}
+        response = self.language_client.analyze_entities(
+            request={"document": document, "encoding_type": language_v2.EncodingType.UTF8}
         )
-        result = ""
-        for entity in response.entities:
-            result += f"{entity.name} ({language_v2.Entity.Type(entity.type_).name}), "
-        language_response = result.rstrip(", ")
-        return language_response
+        result = ', '.join([f"{entity.name} ({language_v2.Entity.Type(entity.type_).name})" for entity in response.entities])
+        return result
+
 
     # TODO  We'll use the GPT-4o mini model, system prompt will come from cognition work
-    def query_gpt(self, query: str, system_prompt: str):
-        response = submit_prompt([
-            system_message(
-                system_prompt
-            ),
-            user_message(query),
-        ])
-        return response
\ No newline at end of file
+    def query_gpt(self, query: str, gpt_system_prompt: str):
+        prompt = [
+            {"role": "system", "content": gpt_system_prompt},
+            {"role": "user", "content": query}
+        ]
+        response = self.openai_client.chat.completions.create(
+            model="gpt-4o-mini",
+            messages=prompt,
+            temperature=0.7,
+        )
+        return response.choices[0].message.content if response.choices else ""
```

Comment:
it would be better to return `None` instead of an empty string (this makes null-checking much easier for whoever/whatever calls the function).